### ISSUE TITLE
Upcoming release update [v1.1]

### ISSUE THREAD

**Issue by dariakhimych** [flyerhq/flutter_chat_ui#24]
For over a month the new features development was blocked by the animated version of chat UI where we faced particular bottlenecks while working with list animations in Flutter. As a result - lack of updates from our side. That's why we are merging issues #7, #9 and #20 into one for a more convenient progress tracking. 

**v1.1 will have 👉**

- [x] Animated chat UI
- [x] Updated date dividers
- [x] Updated message statuses
- [x] Ability to display user name & avatar
- [x] Pagination

💬 From now on all new updates will be shared by @demchenkoalex here.

**Comment by dariakhimych** [comment]
Howdy 👋 Just a small design update for the **group chat**. Will be live soon!

<img width="375" alt="flyer_group_chat" src="https://user-images.githubusercontent.com/33809426/118504107-88b56c80-b72b-11eb-8b13-cf292f90d4c0.png">

**Comment by demchenkoalex** [comment]
And here's an update on animations. Working on a pagination (lazy loading) now and @Voidozzer is looking at user names and avatars.

<img src="https://user-images.githubusercontent.com/14123304/118506389-8fdd7a00-b72d-11eb-81cb-a78112424dde.gif" width="240" height="534" />

**Comment by jangruenwaldt** [comment]
Awesome, great work! Will this all be in one update or will there be several ones?

**Comment by demchenkoalex** [comment]
I guess one, because of breaking changes needed to support names and avatars.

**Comment by m-j-g** [comment]
Nice!

**Comment by vijay-anand-wk** [comment]
Hi @demchenkoalex, is there a scheduled date for the release?

**Comment by dariakhimych** [comment]
@vijay-anand-wk Hey! As far as we do the majority of work over weekends, I would say the release doesn't come earlier than in 2-3 weeks.

**Comment by jlubeck** [comment]
> Howdy 👋 Just a small design update for the **group chat**. Will be live soon!
> 
> <img alt="flyer_group_chat" width="375" src="https://user-images.githubusercontent.com/33809426/118504107-88b56c80-b72b-11eb-8b13-cf292f90d4c0.png">

This is looking very clean, love it! Just a recommendation, User names should have different colors assigned to them to quickly identify when a message is coming from the same person. This is how I solved it on another chat project of mine (looking forward to switching over to this one), but of course feel free to find a better approach!

```dart
Text(user.displayName,
    style: TextStyle(
    color: RandomColor(user.displayName.hashCode).randomColor(colorBrightness: ColorBrightness.dark),
     ),
)
```

`RandomColor` comes from this package: https://pub.dev/packages/random_color

And you can change the `colorBrightness` parameter based on whether the current `ChatTheme` is dark or light to make everything look good.

**Comment by demchenkoalex** [comment]
Weekend update:

- Pagination is done, just a small issue to be resolved before the merge
- Names and avatars are in the PR, to be reviewed
- Names and avatars will probably have different colors for different users, as suggested above

When everything above is merged, new statuses and update to the `ChatTheme` to include more granular settings will be 2 things that left before 1.1

**Comment by demchenkoalex** [comment]
Pagination has been finished and merged to the [v1.1 branch](https://github.com/flyerhq/flutter_chat_ui/tree/feature/animation).

**Comment by demchenkoalex** [comment]
Hi, had a surgery on weekend, still in recovery, will come back to this ASAP

**Comment by jlubeck** [comment]
Hope it was nothing serious!

**Comment by demchenkoalex** [comment]
Nah, some small stuff :) user names and avatars are merged to a feature branch, but there are a couple of small bugs to be fixed. Statuses and theme changes to go and we are finally done with this long awaited release :)

**Comment by jlubeck** [comment]
@demchenkoalex I noticed on the new branch where you guys are working on this (which is looking great btw!) the `shouldRenderTime` property of the Message bubbles is hardcoded to `false`. Is there a reason for that? Is there another way to display message time that I'm not finding in that branch? 

Thanks!

**Comment by demchenkoalex** [comment]
The branch is not ready yet, it will be removed completely. Time is moved to the headers (similar to Facebook Messenger). There will be no time under messages because of the animation issues. In the future we will consider showing time on press (again like in Facebook Messenger) or on swipe (similar to iMessage, Instagram Direct), but it is out of scope for v1.1. Anyway, who needs the exact time when headers will be shown every 15 minutes (again, taken from the Facebook Messenger).

**Comment by jlubeck** [comment]
ok, thank you for clarifying... on very active channels though, a LOT of messages can happen in 15 minutes... I'm running into this issue right now, that's why I brought it up haha. I'm building an app for a group with over 4k members. Wasn't aware that Facebook does that... I use WhatsApp and Telegram and they both include a very small timestamp in the bottom right corner (inside the bubble, so it shouldn't mess with animations?). Will be happy to provide a PR after 1.1 is merged to master if you don't see the value in it, and you can check it out at the time.

**Comment by demchenkoalex** [comment]
Weekend update: finally this is finished. Started as a request to add pagination and ended up as a complete overhaul of the Chat UI, a major breaking change. I expect this to be a one-off action, without huge and breaking refactors in the future, given that I received a lot of good feedback in a couple of months and tried to implement something + left a room for incremental improvements in the future. New version is available on this branch/PR #41. I still need to prepare the documentation as well as upgrade guide, and test the Firebase with the new UI, which hopefully will not take long. After this is done v1.1 will be released on Pub. Thanks everyone.

**Comment by jlubeck** [comment]
> Weekend update: finally this is finished. Started as a request to add pagination and ended up as a complete overhaul of the Chat UI, a major breaking change. I expect this to be a one-off action, without huge and breaking refactors in the future, given that I received a lot of good feedback in a couple of months and tried to implement something + left a room for incremental improvements in the future. New version is available on this branch/PR #41. I still need to prepare the documentation as well as upgrade guide, and test the Firebase with the new UI, which hopefully will not take long. After this is done v1.1 will be released on Pub. Thanks everyone.

Sweeet!! Do you want feedback on that branch if we find anything? Or wait until it's on Pub? Great work!

**Comment by demchenkoalex** [comment]
Released in [v1.1](https://github.com/flyerhq/flutter_chat_ui/releases/tag/v1.1.0).
